src/util/batchInsert.js

```diff
@@ -35,21 +35,23 @@ export default function batchInsert(client, tableName, batch, chunkSize = 1000)
       .then((tr) => {
         return Promise.mapSeries(chunks, (items) => tr(tableName).insert(items, returning))
           .then((result) => {
-            result = flatten(result);
+            result = flatten(result || []);
 
             if(autoTransaction) {
-              return tr.commit()
+              //TODO: -- Oracle tr.commit() does not return a 'thenable' !? Ugly hack for now.
+              return (tr.commit(result) || Promise.resolve())

```

Comment:
Not too familiar with Oracles code base so not sure. The current state came as a shock to me. If `commit` does not return a promise, how can a standard transaction work at all for oracle?

```javascript
knex.transaction((tr) => tr.raw('select version'))
.then(() => {...})
.catch((error) => {})
```

Really strange and confusing. Currently have no environment for testing oracle.